workspace-files/main.go

```diff
@@ -169,6 +173,13 @@ func list(ctx context.Context, filename string) error {
 }
 
 func read(ctx context.Context, filename string) error {
+
+	for _, ext := range unsupportedFileTypes {
```

Comment:
i think this line below
```
	if utf8.Valid(data) {
```
handles the prevention of reading binary files. do we need both?

Reply:
`utf8.Valid(data)` is more like validating  whether the data slice contains only valid UTF-8 encoded characters. it can also handle the case that the input text file is somehow corrupted.

On the other hand, this list of common unsupported binary files is used to filter out unsupported files. 

I decided to go with `unsupported` list because technically  text input file with any non-binary extension (such as .log, .mdtxt, .ipynb, etc) should be allowed, so I guess it's easier to maintain a list of known binary file extensions that we want to filter out.

Reply:
ok. you convinced me.